### ISSUE TITLE
Bug when stopping the collector process: TCP and UDP servers are not closed properly.

### ISSUE THREAD

**Issue by srikartati** [vmware/go-ipfix#219]
**Describe the bug**
When the following extra checks are added in `TestTCPCollectingProcess_ReceiveDataRecord` and `TestUDPCollectingProcess_ReceiveDataRecord` after stopping the collector process, then both the tests fail.

```
cp.Stop()
// Check if connection has closed properly or not by creating a new connection.
_, err = net.Dial(collectorAddr.Network(), collectorAddr.String())
assert.Error(t, err) 
```
```
cp.Stop()
// Check if connection has closed properly or not by creating a new connection.
_, err =  net.DialUDP(udpTransport, nil, resolveAddr)
assert.Error(t, err) 
```

**To Reproduce**
Run the tests by adding above lines

**Expected behavior**
Both TCP and UDP servers should be closed properly after calling `Stop` method of the collector process.

**Actual Behavior**
Servers are not closed properly. We can still create the connections.

**Versions**
ToT main branch

**Additional context**

**Comment by srikartati** [comment]
Reopening the issue as UDP server is not resolved yet.

**Comment by srikartati** [comment]
@zyiou can we close this issue?

**Comment by zyiou** [comment]
> @zyiou can we close this issue?

Yes, the issue has been solved by #227 and #229
